### PR TITLE
New splash screens and deprecation of Mantidplot

### LINKED ISSUE #28719: Add deprecation warning to v5.1 Splash screen
**Original reporter:** Pas

Add an obvious warning to the Splash screen (for plot and WB) for version 5.1 that this is the last version of Mantid that includes MantidPlot!

Also add this to the startup log message!

### PR DESCRIPTION
**Pull request by NickDraper**
**Description of the work**
See commit messages, particularly the second one.

**To test:**
1. Start Mantiplot, look at the mantiplotplot specific "end of life" splash screen
1. Once started there should be a warning level message in the log about Workbench replacing mantidplot
1. The title bar should read "Mantidplot (End of life, use Workbench)", This should survive saving a project, loading a project and starting a new project.
1. Look over the changes to the release notes.
1. Start Workbench, see the workbench specific splash screen and no end of life message, also no end of life message in the log on startup.

Fixes #28719

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: docs/source/release/v5.1.0/ui.rst [hunk 1]
```diff
@@ -19,7 +19,6 @@ See :doc:`mantidplot`.
 
 MantidWorkbench
 ---------------
```
Comment (OwenArnold) [inline docs/source/release/v5.1.0/ui.rst:21]: Are you using `MantidWorkbench` `workbench`, `Workbench`? `MantidPlot` has been a fairly consistently used for comparison, but there are a range of usages of the former in this PR.

File: MantidPlot/src/main.cpp [hunk 2]
```diff
@@ -207,7 +207,7 @@ int main(int argc, char **argv) {
   try {
     // Splash
     QPixmap pixmap;
-    if (!pixmap.load(":/MantidSplashScreen.png"))
+    if (!pixmap.load(":/MantidPlotSplashScreen-Deprecated.png"))
```
Comment (OwenArnold) [inline MantidPlot/src/main.cpp:210]: End of life does not specify a date easily to hand. Would it be better to say something like `final stable release` or provide a date?

**Review by OwenArnold — COMMENTED** [review]
The splash screens use `Workbench` which is different case to `mantid` in text above and appears to be a different font. It is also left aligned rather than centred.

File: MantidPlot/src/main.cpp (hunk 2 above)
Comment (NickDraper) [reply MantidPlot/src/main.cpp:210]: We told by the PMB not to specify a date, as they were concerned that is would delay scientists moving over until the date is imminent

However we need to make it prominent that action to move over is important and final stable release does not really suggest that.  The PMB want it to be very clear on the splash screen that this  is the end.

File: docs/source/release/v5.1.0/ui.rst (hunk 1 above)
Comment (NickDraper) [reply docs/source/release/v5.1.0/ui.rst:21]: I agree we should remove refences to MantidWorkbench, as it has never been that it is just a term that has gotten into our release notes.

**Comment by NickDraper** [comment]
> The splash screens use `Workbench` which is different case to `mantid` in text above and appears to be a different font. It is also left aligned rather than centered.

The Font difference is unavoidable as we do not have easy access to the original font.  I considered centered but preferred aligned to the left myself.

**Comment by martyngigg** [comment]
As this is going into the nightly are we doing this a little early? Given it doesn't mention a last release number might we confuse people downloading the nightly?
Can we mention 5.1 being the last release shipping MantidPlot?

**Comment by DanNixon** [comment]
This is a slight tangent, but have we considered if we actually want to keep the splash screen in Workbench?
They are a bit of an outdated concept (I could only find two things I frequently use that still show them) and unless we show startup progress then they are rather useless (unless the facilities and Tessella desperately care about self promotion).

**Comment by NickDraper** [comment]
> This is a slight tangent, but have we considered if we actually want to keep the splash screen in Workbench?
> They are a bit of an outdated concept (I could only find two things I frequently use that still show them) and unless we show startup progress then they are rather useless (unless the facilities and Tessella desperately care about self promotion).

Workbench does take a while to start, and we have feedback from our users that they need to see the splash screen in order to know that the program is actually starting so they don't try to start it again.  We are looking at the startup speed, but it is very unlikely we will get it fast enough not to need this.

**Comment by NickDraper** [comment]
> As this is going into the nightly are we doing this a little early? Given it doesn't mention a last release number might we confuse people downloading the nightly?
> Can we mention 5.1 being the last release shipping MantidPlot?

I'll add in the version number on the log message.  We also have this message on the home page and forum, so there are plenty of places people can get more information.

**Comment by OwenArnold** [comment]
> > This is a slight tangent, but have we considered if we actually want to keep the splash screen in Workbench?
> > They are a bit of an outdated concept (I could only find two things I frequently use that still show them) and unless we show startup progress then they are rather useless (unless the facilities and Tessella desperately care about self promotion).
> 
> Workbench does take a while to start, and we have feedback from our users that they need to see the splash screen in order to know that the program is actually starting so they don't try to start it again. We are looking at the startup speed, but it is very unlikely we will get it fast enough not to need this.

Maybe more a question for @martyngigg can we reliably produce progress information for start times, because if we can it would be nice to have it on the splash as @DanNixon mentioned.

**Comment by NickDraper** [comment]
Perhaps, but out of scope for this PR, which is about the deprecation message for Mantidplot.

**Comment by NickDraper** [comment]
Created #28757 to update the release notes generator for future releases

**Review by AndreiSavici — COMMENTED** [review]
Works as expected (mostly). Can we get the same message in the First Time Setup dialog box and in the About Mantidplot one?

**Review by DanielMurphy22 — CHANGES_REQUESTED** [review]
I think it's hard to know how much is enough without being blinding. I like having `MantidPlot (End of life, use Workbench)` on the topbar! If the splash screen is relatively fast, then the message could be missed and the results log is easily ignored. 
I don't think the Help About menu is neccesary, but I think adding the words from the splash screen to the First Time Setup window would be useful.

**Review by DanielMurphy22 — APPROVED** [review]
Looks good to me!

**Pull request merged**